inc/plugin/hookmanager.class.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Plugin;
+
+class HookManager
+{
+   protected string $plugin;
+
+   public function __construct(string $plugin) {
+      $this->plugin = $plugin;
+   }
+
+   /**
+    * Enable CSRF
+    */
+   public function enableCSRF(): void {
+      $PLUGIN_HOOKS[Hooks::CSRF_COMPLIANT][$this->plugin] = true;
+   }
+
+   /**
+    * Add a given javascript file
+    *
+    * @param string $file
+    */
+   public function registerJavascriptFile(string $file): void {
```

Comment:
Needs to support `add_javascript_module` hooks in addition to the plain `add_javascript` hook.

Reply:
I can't find any references to `add_javascript_module` in the code.
Are you sure it's available on this branch ?

Reply:
My bad. #8970 targeted the modern-ui branch and not master so it isn't in the master branch yet.